scipy/optimize/_spectral.py

```diff
@@ -58,7 +59,11 @@ def _root_df_sane(func, x0, args=(), ftol=1e-8, fatol=1e-300, maxfev=1000,
     M : int, optional
         Number of iterates to include in the 'cruz' nonmonotonic line search.
         Default: 10
-    line_search : {'cruz', 'cheng'}
+    stagnation_limit : int, optional
+        If merit function does not decrease in `stagnation_limit` iterations,
+        terminate with failure.
+        Default: 100
```

Comment:
I've seen this option being called "patience" in other contexts (backpropagation in particular). Wouldn't that be a better name, it being easier to type?